subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestEventSerializer.java

```diff
@@ -183,12 +182,19 @@ public DefaultTestFailure read(Decoder decoder) throws Exception {
             for (int i = 0; i < numOfCauses; i++) {
                 causes.add(read(decoder));
             }
+
+            String rawFailureName = decoder.readString();
+            Throwable rawFailure;
+            try {
+               rawFailure = throwableSerializer.read(decoder);
+            } catch(Exception e) {
+                rawFailure = new IllegalStateException("An exception of type " + rawFailureName + " was thrown by the test, but there was a failure while transmitting the exception back to the build process", e);
```

Comment:
❔ Is "while transmitting" a proper description?  This makes me think of a communication/networking issue, not a serialization issue.  Also "there was a failure" should probably be rewritten to use active voice.

Reply:
I changed this to "but Gradle was unable to recreate the exception in the build process".  I'm struggling to describe this in a way that is sensible to the user without being too abstract (i.e. I don't think the user knows or cares if the exception is being serialized/deserialized so I wanted to avoid being that technical).  I don't feel like the current message is great, either, but I couldn't think of anything better.  I'm open to suggestions. if you have better ideas.

Reply:
I think that's already a big improvement.  Should be enough of a breadcrumb, and I don't have anything better.